src/core/lib/promise/activity.h

```diff
@@ -387,8 +413,16 @@ class PromiseActivity final
         return IntoStatus(status);
       }
       // Continue looping til no wakeups occur.
-    } while (got_wakeup());
-    return {};
+      switch (got_action_during_run()) {
+        case ActionDuringRun::kNone:
+          return {};
+        case ActionDuringRun::kWakeup:
+          break;
+        case ActionDuringRun::kCancel:
+          MarkDone();
+          return absl::CancelledError();
+      }
+    };
```

Comment:
Does this pass a `-Wunreachable-code-aggressive` check? If not, the chromium folks may have issues.

Reply:
I expect so? It's nested in a while loop, so if we fall out of the switch we'll just keep spinning.

Reply:
Uf, my mistake. I was actually seeing potential for the opposite, a possible missing return, but that's not a problem either.